source/SIL.AppBuilder.Portal.Frontend/src/sockets/data-socket.tsx

```diff
@@ -109,27 +115,34 @@ function handleSocketPayload(dataStore: Store, json: string) {
   return response;
 }
 
-function useConnectionStateWatcher(connection: HubConnection<DataHub>) {
-  const subscription = useRef<Subscription>();
+function useConnectionStateWatcher(dataClient: DataSocketClient) {
+  const connectionSubscription = useRef<Subscription>();
+  const stateSubscription = useRef<Subscription>();
   const [isConnected, setIsConnected] = useState(false);
   const [connectionState, setConnectionState] = useState<ConnectionState>();
 
   useEffect(() => {
     if (isTesting) return;
 
-    if (!subscription.current) {
-      subscription.current = connection.connectionState$.subscribe((state) => {
-        setIsConnected(state.status === ConnectionStatus.connected);
-        setConnectionState(state);
+    if (!connectionSubscription.current) {
```

Comment:
this change is "wait for the socket to become connected before trying to observe the connection state".

I don't know how much of a difference this makes.